app/src/main/java/agency/five/codebase/android/movieapp/ui/component/MovieCategoryLabel.kt

```diff
@@ -0,0 +1,91 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import agency.five.codebase.android.movieapp.R
+import androidx.annotation.StringRes
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.Divider
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+data class MovieCategoryLabelViewState(
+    val itemId: Int,
+    val isSelected: Boolean,
+    val categoryText: MovieCategoryLabelTextViewState
+)
+
+sealed class MovieCategoryLabelTextViewState {
+    class LabelTextFromString(val text: String) : MovieCategoryLabelTextViewState()
+    class LabelTextFromResource(@StringRes val textRes: Int) : MovieCategoryLabelTextViewState()
+}
+
+@Composable
+fun MovieCategoryLabel(
+    labelViewState: MovieCategoryLabelViewState
+) {
```

Comment:
This should be clickable. lambda should be passed to MovieCategoryLabel as parameter.
I think it is great that you created 2 composables ( 1 for each state), but because you have to add clickable functionality I suggest you extract Column to MovieCategoryLabel where you will define size (as you did ) and add clickable modifier, and everything else can be defined in separate methods as you did.